llvm/test/CodeGen/AArch64/complex-deinterleaving-cdot.ll

```diff
@@ -0,0 +1,58 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S --passes=complex-deinterleaving %s --mattr=+sve2 -o - | FileCheck %s
```

Comment:
We need some more tests, such as:
* No sve, no sve2
* More rotations
* I don't really understand the caching system, but some tests that test that functionality are needed
* More types

And negative tests for other code paths.